### PR TITLE
fix: parallelism should not be subtracted 1

### PR DESCRIPTION
**Pull request by fuyufjh**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

As title.

For example, given `unsafe_worker_node_parallel_degree = 4`, there will be 4 hash parallel units and 1 single parallel units.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Review by BugenZhao — APPROVED** [review]

File: src/config/risingwave.toml [hunk 1]
```diff
@@ -6,7 +6,7 @@ heartbeat_interval_ms = 1000
 [streaming]
 checkpoint_interval_ms = 250
 in_flight_barrier_nums = 40
-unsafe_worker_node_parallel_degree = 2
```
Comment (BugenZhao) [inline src/config/risingwave.toml:9]: My fault. 🥵

**Comment by skyzh** [comment]
I'd prefer to set it to 3, so that benchmarks will be consistent.

**Comment by skyzh** [comment]
Also `unsafe_worker_node_parallel_degree` should not appear in the configuration. Only developers need to know it. 😇🥵

**Comment by fuyufjh** [comment]
> I'd prefer to set it to 3, so that benchmarks will be consistent.

Yes I've also considered this, but our benchmark will continue for months, and I think it's better to fix it now.

**Comment by skyzh** [comment]
> > I'd prefer to set it to 3, so that benchmarks will be consistent.
> 
> Yes I've also considered this, but our benchmark will continue for months, and I think it's better to fix it now.

Alright.

**Review by skyzh — APPROVED** [review]

**Comment by BugenZhao** [comment]
```
| \| e2e_test/streaming/./tpch/views/q19.slt.part                 .. [OK] in 8649 ms
| \| e2e_test/streaming/./tpch/views/q20.slt.part                 .. [OK] in 13466 ms
| \| e2e_test/streaming/./tpch/views/q22.slt.part                 .. [OK] in 11529 ms
```

e2e become much slower than before. That's interesting. 🤔

**Pull request merged**